src/olympia/bandwagon/views.py

```diff
@@ -676,16 +674,27 @@ def get_queryset(self):
             author=self.get_account_viewset().get_object())
 
 
-class CollectionAddonViewSet(ListModelMixin, GenericViewSet):
-    permission_classes = [AllowAny]
+class CollectionAddonViewSet(ModelViewSet):
+    permission_classes = []  # We don't need extra permissions.
     serializer_class = CollectionAddonSerializer
+    lookup_field = 'addon'
 
-    def get_queryset(self):
-        if not hasattr(self, 'collection_object'):
-            self.collection_object = CollectionViewSet(
+    def get_collection_viewset(self):
+        if not hasattr(self, 'collection_viewset'):
+            # CollectionViewSet's permission_classes are good for us.
+            self.collection_viewset = CollectionViewSet(
                 request=self.request,
                 kwargs={'user_pk': self.kwargs['user_pk'],
-                        'slug': self.kwargs['collection_slug']}).get_object()
+                        'slug': self.kwargs['collection_slug']})
+        return self.collection_viewset
+
+    def get_object(self):
+        self.lookup_url_kwarg = self.lookup_url_kwarg or self.lookup_field
+        # if the lookup is not a number, its probably the slug instead.
+        if not self.kwargs[self.lookup_url_kwarg].isdigit():
```

Comment:
Generally, it's not explicitly passed as a string in Detail or Patch [tests](https://github.com/mozilla/addons-server/pull/5563/files#diff-87decf4b1acc3b95c24f8bcb9c4bdc0aR1720) - just `self.addon.id` and it seems to work (I believe because it's encoded in the url, and only used for lookup, not create where the `.data` might contain non-string data types).

If it was missing entirely the url regex wouldn't be matched and the view wouldn't be called - `lookup_url_kwarg` is used by the DRF Router to name the part after `collections` in the url.